assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+            }
+            else{
+                this.error_code = 2;
+                this.event_executor('error');
+            }
+            if(option.startSeconds!==undefined&&typeof option.startSeconds==='number'&&option.startSeconds>=0){
+                startSeconds = option.startSeconds;
+            }
+            if(option.endSeconds!==undefined&&typeof option.endSeconds==='number'&&option.endSeconds>=0){
+                endSeconds = option.endSeconds;
+            }
+        }
+        if(mediaContetUrl.length>0){
+            var tmp_videoId = '';
+            if(mediaContetUrl.indexOf('/v/')!==-1){
+                var end_pos = mediaContetUrl.length-1;
+                if(mediaContetUrl.indexOf('?')!==-1){
+                    end_pos = mediaContetUrl.indexOf('?');
+                }
+                tmp_videoId = mediaContetUrl.substring(mediaContetUrl.indexOf('/v/'),end_pos);
+            }
+            else{
+                tmp_videoId = new URL(mediaContetUrl).searchParams.get('v');
+            }
+            if(tmp_videoId===null||tmp_videoId.length!==11){
+                this.error_code = 2;
+                this.event_executor('error');
+            }
+            videoId = tmp_videoId;
+        }
+        var iframe_sorce = this.target_origin.slice();
+        iframe_sorce += "/embed/" + videoId;
+        this.videoId = videoId;
+        var search_params = new URLSearchParams('');
+        search_params.append('origin',location.origin);
+        search_params.append('enablejsapi','1');
+        search_params.append('widgetid',invidious_embed.widgetid);
+        this.widgetid = invidious_embed.widgetid;
+        invidious_embed.widgetid++;
+        if(autoplay){
+            search_params.append('autoplay',1);
+        }
+        else{
+            search_params.append('autoplay',0);
+        }
+        if(this.option_playerVars!==undefined){
+            for(var x in this.option_playerVars){
+                if(x!=='autoplay'&&x!=='start'&&x!=='end'){
+                    search_params.append(x,this.option_playerVars[x]);
+                }
+            }
+        }
+        if(startSeconds!==-1&&startSeconds>=0){
+            search_params.append('start',startSeconds);
+        }
+        if(endSeconds!==-1&&endSeconds>=0){
+            if(endSeconds>startSeconds){
+                search_params.append('end',endSeconds);
+            }
+            else{
+                throw 'invalid end seconds';
+            }
+        }
+        iframe_sorce += "?" + search_params.toString();
+        this.player_iframe.src = iframe_sorce;
+        if(autoplay){
+            this.player_status = 5;
+        }
+        this.eventdata = {};
+    }
+    loadVideoById(option,startSeconds){
+        this.playOtherVideoById(option,true,startSeconds);
+    }
+    cueVideoById(option,startSeconds){
+        this.playOtherVideoById(option,false,startSeconds);
+    }
+    cuevVideoByUrl(option,startSeconds){
```

Comment:
```suggestion
    cueVideoByUrl(option,startSeconds){
```
How this was tested? Additionally, cueVideoByUrl accept third argument `suggestedQuality`.